### ISSUE TITLE
Cannot create route with authentication for custom plugin

### ISSUE THREAD

**Issue by giangvo1125** [medusajs/medusa#4683]
### Describe the bug

Hello, I have an issue when creating my custom plugin. I follow [this](https://docs.medusajs.com/development/plugins/create) to create the custom plugin but I cannot define an admin route using authentication middleware to authenticate my route. Its throw `Error: Unknown authentication strategy "admin-jwt"`. 

### System information

Medusa version (including plugins): 1.12.0
Node.js version: 16.20.0
Database: Postgres v15.3-alpine3.18
Operating system: MacOS
Browser (if relevant):

### Expected behavior

Can use the authenticate middleware for the admin route defined by the plugin.

### Screenshots
<img width="771" alt="Screenshot 2023-08-03 at 15 25 56" src="https://github.com/medusajs/medusa/assets/11868731/b018302e-1ed3-467a-9c24-e4c1856a3c7d">

<img width="840" alt="Screenshot 2023-08-03 at 15 26 09" src="https://github.com/medusajs/medusa/assets/11868731/96359ee5-fcc6-46ef-becc-bc405fe03f19">

**Comment by quang-ng** [comment]
We are encountering the same problem. Kindly notify us if there are any updates, @giangvo1125

**Comment by quangnst** [comment]
I'm also having this problem, hope it gets resolved

**Comment by pevey** [comment]
On the plugin guide page you linked, scroll way down to the troubleshooting section, and do the yarn linking steps listed:

```bash
cd <BACKEND_PATH>/node_modules/medusa-interfaces
yarn link
cd <BACKEND_PATH>/node_modules/@medusajs/medusa
yarn link
cd <PLUGIN_PATH>
rm -rf node_modules/medusa-interfaces
rm -rf node_modules/@medusajs/medusa
yarn link medusa-interfaces
yarn link @medusajs/medusa
yarn link
cd <BACKEND_PATH>
yarn link your-plugin
```

What is happening is that your medusa project is installing the plugin, which in turn already has @medusajs/medusa as dependency in its node_modules folder.  So there are two versions of medusa.  And only one is the one you want (the top level project copy).  So, your plugin is using an instance of passport that does not have the auth strategy defined.

The linking procedures above will fix this when you are developing a plugin locally. 

EDIT: While the troubleshooting note specifically points out medusa-interfaces and @medusajs/medusa, it is best to do this for any medusa package you need to import from in your package.  That could include medusa-core-utils, @medusajs/utils, etc.

**Comment by giangvo1125** [comment]
Thanks @pevey for the reply. I did like this but it's not working for now. And I think it may not be a problem because we use the `link` command when we test locally. How can we do this when publishing the plugin?

**Comment by pevey** [comment]
Did you follow the steps exactly?  All of them?  Using link to install the plugin locally does not solve the problem.  In fact, that is the root of the problem.  You will not get the same issue when you publish the plugin to npm and install using a package manager. 

If you post your code in the Help section of the Medusa discord, there are more people there who may be able to help you troubleshoot it.  

But it is not a bug.  It is just the nature of importing from a dependency that is also at the parent level of the project.  It is the nature of having two copies of a package installed.  CJS does not handle it well.  If you install from a package manager, there are not existing copies of the medusa packages in the plugin's node modules folder, and extra copies do not get installed since npm can see that they are already present.  But when developing locally but not in a monorepo, you have extra copies.

See https://github.com/pevey/medusa-auth-cognito for an example that it does work.

**Comment by giangvo1125** [comment]
@pevey Yes, I follow exactly the steps. And the problem for me is I cannot create a route using authentication from the admin instead of creating a custom authentication.

**Comment by pevey** [comment]
I understand, but the example plugin uses similar functions from the same service.  You can also see Medusa-plugins-ratings if you want to see specially an admin route. The process is the same.  Please post your code in the discord. People can help you with troubleshooting there. The issues tracker is meant for reporting bugs, so the volunteers who can help you with this aren’t here as much as they are in discord.  If you search the docs, there is also a guide for creating custom routes with authentication. Try creating the route within the Medusa project folder first, and only after it is working, tackle the challenge of moving it into a plugin.

**Comment by sradevski** [comment]
Hey, thanks for the report! Since v2 brought a lot of architectural and API changes on the backend, we will be closing this ticket since it no longer applies to our new setup, or the issue has already been fixed. If you are still facing issues with v1,
please open a new ticket and we will address it as soon as possible. Thanks! 🙏
